fx-src/FirefoxStorage.js

```diff
@@ -144,10 +148,10 @@ export default class FirefoxAdapter extends BaseAdapter {
   open() {
     const self = this;
     return Task.spawn(function* (){
-      const path = self._options.path || SQLITE_PATH;
-      const opts = { path, sharedMemoryCache: false };
       if (!self._connection) {
-        self._connection = yield Sqlite.openConnection(opts).then(self._init);
+        const path = self._options.path || SQLITE_PATH;
+        const opts = { path, sharedMemoryCache: false };
+        self._connection = yield Sqlite.openConnection(opts).then(FirefoxAdapter._init);
```

Comment:
I'm reminded of the old Rusty Russell blog posts about API design ([intro](http://ozlabs.org/~rusty/index.cgi/tech/2008-03-18.html), [part 1](http://ozlabs.org/~rusty/index.cgi/tech/2008-03-30.html), [part 2](http://ozlabs.org/~rusty/index.cgi/tech/2008-04-01.html)). I'm nervous about this API being easy to misuse. The original API *requires* you to call `open` or else it always breaks. The new one allows you to pass in an uninitialized Sqlite handle, and it *might* break depending on the previous state of the DB. I'd like the "default" interface to be safe, even if some people with special needs can avoid it if they need to. Maybe setting an attribute on a SqliteConnection that says it's been initialized would be a good idea?